### PR TITLE
Vald V1 New Design APIs

### PR DESCRIPTION
**Pull request by kpango**
Signed-off-by: kpango <i.can.feel.gravity@gmail.com>

### Description:
This PR include Vald's new design APIs
- proto file
- Makefile / Makefile.d/*
- generated code such as grpc, swagger, api documents

This change includes a lot of features.
I'll make sub PRs (I'll add links below) for this branch please review each PRs.
- test asset changes #741 
- internal package changes #742
     - agent changes #743
     - manager changes #744
     - meta gateway #746 
     - lb gateway #747 
     - backup gateway #748 
     - manifests #749

### Environment:

- Go Version: 1.15.4
- Docker Version: 20.10.0-beta1
- Kubernetes Version: 1.18.9
- NGT Version: 1.12.2

### Types of changes:

- [ ] Bug fix [type/bug]
- [x] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/gateway/lb/handler/grpc/handler.go [hunk 1]
```diff
@@ -0,0 +1,994 @@
...
+
+const apiName = "vald/gateway-lb"
+
+func New(opts ...Option) vald.Server {
+	s := new(server)
+
+	for _, opt := range append(defaultOptions, opts...) {
+		opt(s)
+	}
+	return nil
```
Comment (rinx) [inline pkg/gateway/lb/handler/grpc/handler.go:59]:
```suggestion
	return s
```

Comment (kpango) [reply pkg/gateway/lb/handler/grpc/handler.go:59]: thanks

File: pkg/gateway/lb/handler/grpc/handler.go [hunk 2]
```diff
@@ -0,0 +1,994 @@
...
+		}
+		if ovec != nil && ovec.GetId() != "" && ovec.GetVector() != nil {
+			once.Do(func() {
+				vec = ovec
+				cancel()
+			})
+		}
+		return nil
+	})
+	if err != nil || vec == nil || vec.GetId() != "" || vec.GetVector() != nil {
```
Comment (rinx) [inline pkg/gateway/lb/handler/grpc/handler.go]: maybe
```suggestion
	if err != nil || vec == nil || vec.GetId() == "" || vec.GetVector() == nil {
```

**Comment by kpango** [comment]
@vdaas/set @vdaas/core @vdaas/sre Sorry for in inconvenience for a long time and sorry for making bad & huge PR.
I'm going to merge this  PR. thank you for your cooperations.

**Pull request merged**